infrastructure/src/main/java/org/corfudb/infrastructure/BatchWriter.java

```diff
@@ -122,6 +129,8 @@ private void batchWriteProcessor() {
                         streamLog.append(currOp.getLogAddress(), currOp.getLogData());
                         currOp.setException(null);
                         res.add(currOp);
+                        long globalAddress = currOp.getLogAddress().getAddress();
```

Comment:
Yeah, I implemented it this way earlier. Its just that we would need an AtomicLong to maintain and access this variable atomically.